docs/scripts/buildApi.ts

```diff
@@ -30,11 +30,32 @@ function generateType(type, needsParenthesis = false) {
   }
   if (type.type === 'reflection') {
     if (type.declaration.signatures && type.declaration.signatures.length === 1) {
-      const signature = type.declaration.signatures[0];
+      const signature: TypeDoc.SignatureReflection = type.declaration.signatures[0];
+
       let text = needsParenthesis ? '(' : '';
+
+      // Handle function generic parameters
+      if (signature.typeParameters?.length) {
+        text += '<';
+        text += signature.typeParameters.map((generic) => {
+          let genericLine = generic.name;
+
+          if (generic.type) {
+            genericLine += ` extends ${generateType(generic.type)}`;
+          }
+
+          if (generic.default) {
+            genericLine += ` = ${generateType(generic.default)}`;
+          }
+
+          return genericLine;
```

Comment:
We can do the same with less lines:

```suggestion
          let genericLine = generic.name;
          genericLine += generic.default ? ' =' : ' extends';
          return `${genericLine} ${generateType(generic.default || generic.type)}`;
```

Reply:
Yeah, I see. If you like we can use the one-line if syntax: `if (something) genericLine  += '...';` Just to keep all branches small.

Reply:
Done :+1: